core/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java

```diff
@@ -254,8 +255,8 @@ private void restore(final IndexShard indexShard, final IndexShardRepository ind
             translogState.totalOperationsOnStart(0);
             indexShard.prepareForIndexRecovery();
             ShardId snapshotShardId = shardId;
-            if (!shardId.getIndex().equals(restoreSource.index())) {
-                snapshotShardId = new ShardId(restoreSource.index(), shardId.id());
+            if (!shardId.getIndexName().equals(restoreSource.index())) {
+                snapshotShardId = new ShardId(restoreSource.index(), IndexMetaData.INDEX_UUID_NA_VALUE, shardId.id());
```

Comment:
Good question. I wasn’t sure of the exact semantics - we do change the index name, at least as far as identity is now, it’s different for many things so I figured it’s “safer” to indicate this as an I don’t know UUID…

> On 27 Jan 2016, at 14:05, Simon Willnauer <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java:
> 
> > @@ -254,8 +255,8 @@ private void restore(final IndexShard indexShard, final IndexShardRepository ind
> >              translogState.totalOperationsOnStart(0);
> >              indexShard.prepareForIndexRecovery();
> >              ShardId snapshotShardId = shardId;
> > -            if (!shardId.getIndex().equals(restoreSource.index())) {
> > -                snapshotShardId = new ShardId(restoreSource.index(), shardId.id());
> > +            if (!shardId.getIndexName().equals(restoreSource.index())) {
> > +                snapshotShardId = new ShardId(restoreSource.index(), IndexMetaData.INDEX_UUID_NA_VALUE, shardId.id());
> 
> should we preserve the UUID from the incoming shard?
> 
> —
> Reply to this email directly or view it on GitHub.
>